walk-in-interview/src/main/webapp/homepage/script.js

```diff
@@ -0,0 +1,233 @@
+    const sortOrderParam = document.getElementById('homepage-sort-by-order');
+    addJobListings(sortByParam, sortOrderParam, DEFAULT_PAGE_SIZE,
+        DEFAULT_PAGE_INDEX);
+  });
+}
+
+/**
+ * Add the attributes and on click function to the filters
+ * submit button.
+ */
+function addJobFilterSubmit() {
+  const filterBySubmit = document.getElementById('homepage-filter-by-submit');
+  filterBySubmit.setAttribute('type', 'submit');
+  filterBySubmit.setAttribute('value', STRINGS['homepage-filter-by-submit']);
+
+  /**
+   * TODO(issue/35): add filtering event listener
+   * + validation + GET request functions
+   */
+}
+
+/**
+ * Add the list of jobs that are stored in the database.
+ * @param {String} sortBy How the jobs should be sorted.
+ * @param {String} order The order of the sorting.
+ * @param {int} pageSize The number of jobs for the page.
+ * @param {int} pageIndex The page index (starting from 0).
+ */
+function addJobListings(sortBy, order, pageSize, pageIndex) {
+  const jobListings = getJobListings(sortBy, order, pageSize, pageIndex);
+  console.log('jobListings', jobListings);
+  const jobListingsElement = document.getElementById('job-listings');
+
+  jobListingsElement.innerHTML = '';
+  const jobListingTemplate = document.getElementById('job-listing-template');
+
+  jobListings.forEach((job) => {
+    const jobListing =
+      jobListingTemplate.cloneNode( /** and child elements */ true);
+    jobListing.setAttribute('id', job['jobId']);
+
+    const jobTitle = jobListing.children[0];
+    jobTitle.innerText = job['jobTitle'];
+
+    const jobAddress = jobListing.children[1];
+
+    const location = job['jobLocation'];
+    const address = location['address'];
+    const postalCode = location['postalCode'];
+    jobAddress.innerText = address + ', ' + postalCode;
+
+    jobListingsElement.appendChild(jobListing);
+  });
+}
+
+/**
+ * Makes GET request to retrieve all the job listings from the database
+ * given the sorting and order. This function is called when the
+ * homepage is loaded and also when the sorting is changed.
+ * @param {String} sortBy How the jobs should be sorted.
+ * @param {String} order The order of the sorting.
+ * @param {int} pageSize The number of jobs for the page.
+ * @param {int} pageIndex The page index (starting from 0).
+ * @return {Array} response from the GET request with the list of jobs.
+ */
+function getJobListings(sortBy, order, pageSize, pageIndex) {
+  // TODO(issue/18): get jobs from database and render them on screen
+  // returning some hardcoded data for now
+  const params = 'sortBy=' + sortBy +
+    '&order=' + order +
+    '&pageSize=' + pageSize +
+    '&pageIndex=' + pageIndex;
+
+  fetch('/jobs?' + params)
+      .then((response) => response.text())
+      .then((data) => {
+        console.log('data', data);
+        setErrorMessage('', /** include default msg */ false);
+        return data['jobList'];
```

Comment:
here seems `data` is an array? but `response.text()` seems return string

Reply:
I believe `data` is an object, and `data['jobList']` will return the array of objects under that key.

Reply:
if i am not wrong, `data` is the `response.text()`, but `response.text()` returns a string? if want an object, maybe `response.json()`?

Reply:
changed. thanks